src/string.cr

```diff
@@ -1420,8 +1420,8 @@ class String
   # is not included in the *hash*, nothing is replaced.
   #
   # ```
-  # "hello".sub(/(he|l|o)/, {"he": "ha", "l": "la"}) # => "hallo"
-  # "hello".sub(/(he|l|o)/, {"l": "la"})             # => "hello"
+  # "hello".sub(/(he|l|o)/, {"he" => "ha", "l" => "la"}) # => "hallo"
```

Comment:
@Sija 
Yep, I found that too.  `gsub` accepts it, but `sub` does not.

```diff
-  def sub(pattern : Regex, hash : Hash(String, _))
+  def sub(pattern : Regex, hash : Hash(String, _) | NamedTuple)
     sub(pattern) { |match|
-      if hash.has_key?(match)
+      if hash[match]?
```
It seems this is a simple fix, but I'm afraid that `Hash#[]?` would be a bit slower than `Hash#has_key?`. Otherwise, should we implement `NamedTuple#has_key?`?